src/nextgenMissingImageResolutions/index.ts

```
@@ -0,0 +1,116 @@
+  await loadEnv([]);
+  setup();
+  const resolutions = ['2k'];
+  for (let resolution of resolutions) {
+    await findMissingImages(resolution);
+  }
+  const diff = start.diffFromNow().formatAsDuration();
+  logger.info(`[COMPLETE IN ${diff}]`);
+};
+
+function getNetworkPath() {
+  if (process.env.NEXTGEN_CHAIN_ID === sepolia.id.toString()) {
+    return `testnet`;
+  }
+  return `mainnet`;
+}
+
+async function findMissingImages(resolution: string) {
+  const networkPath = getNetworkPath();
+  const resolutionPath = `${networkPath}/png${resolution}/`;
+
+  const allExisting = await listS3Objects(s3, resolutionPath);
+
+  logger.info(
+    `[RESOLUTION ${resolution.toUpperCase()}] : [RESOLUTON_PATH ${resolutionPath}] : [CURRENT IMAGE COUNT ${
+      allExisting.length
+    }]`
+  );
+
+  const nextBatch = await getNextBatch(
+    allExisting,
+    START_INDEX,
+    END_INDEX,
+    BATCH_SIZE
+  );
+
+  if (nextBatch.length) {
+    await uploadBatch(nextBatch, resolutionPath, resolution);
+  } else {
+    logger.info(`[NO MISSING IMAGES]`);
+  }
+}
+
+async function uploadBatch(batch: number[], path: string, resolution: string) {
+  logger.info(
+    `[UPLOADING BATCH] : [RESOLUTION: ${resolution.toUpperCase()}] : [BATCH ${JSON.stringify(
+      batch
+    )}]`
+  );
+  const promises = [];
+  for (let item of batch) {
+    promises.push(uploadMissingNextgenImage(item, resolution));
+  }
+  await Promise.all(promises);
+}
+
+async function uploadMissingNextgenImage(tokenId: number, resolution: string) {
+  const generatorPath = `${GENERATOR_BASE_PATH}/mainnet/png/${tokenId}/${resolution}`;
+  const s3Path = `mainnet/png${resolution}/${tokenId}`;
+
+  logger.info(
+    `[TOKEN_ID ${tokenId}] : [RESOLUTION ${resolution.toUpperCase()}] : [GENERATOR PATH ${generatorPath}] : [S3 PATH ${s3Path}]`
+  );
+
+  const imageBlob = await getImageBlobFromGenerator(generatorPath);
+  if (!imageBlob) {
+    logger.info(`[IMAGE BLOB ERROR] : [EXITING]`);
```

You are catching errors in getImageBlobFromGenerator and info logging them here. Why not just propagate the error up to the top so we could see it in Sentry?

---

i change to logger.error 😬 

i expect that this will fail a lot - and by a lot i mean 5-10% of requests so i think i'll be spamming
its not an error that this logic made, its another service failing to give it a response
and i want the batch to carry on